homeassistant/components/switchbot_cloud/switch.py

```diff
@@ -95,4 +97,12 @@ def _async_make_entity(
         "Relay Switch 1",
     ]:
         return SwitchBotCloudRelaySwitchSwitch(api, device, coordinator)
+    if "Bot" in device.device_type:
+        if (
+            coordinator.data is not None
+            and coordinator.data.get("deviceMode") == "pressMode"
+        ):
+            # pressMode Bots are handled by the Button platform
+            return None
+        return SwitchBotCloudSwitch(api, device, coordinator)
```

Comment:
Why is this here? The switch platform doesn't create things for the Bot right?

Reply:
Bots can be configured to behaved in two different modes:
- in `switchMode`, there are two actions, `on` (the little arm goes out and stays out) and `off` (the little arm goes back in and stays in)
- in `pressMode`, there is only one action: the little arm comes out and immediately goes back in.

(there's a third mode `customizeMode` that has the same actions as `switchMode` but the behavior is customizable).

I've mapped the `switchMode`/`customizeMode` to the switch platform and the `pressMode` to the button platform.

Reply:
A bot can only be in one state at a time?

Reply:
Yes, a bot is only in one mode at a time.

Reply:
The way I've written this code was to get around the lack of access to the first refresh when setting up the devices. I think a refactor of `make_device_data` and associated logic is in order to be able to decided whether a bot is a switch or a button without having to do so in `async_setup_entry`.

Reply:
Here's the PR for the aforementioned refactor: #135698. With that change, the logic to decide how to treat a Bot based on its `deviceMode` can be moved to `make_device_data`, and allowing the complexity I've added to the platforms to be removed